objc/include/onnxruntime/ort_session.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+@class ORTEnv;
+@class ORTValue;
+
+// TODO
+@interface ORTSessionOptions : NSObject
+@end
+
+// TODO
+@interface ORTRunOptions : NSObject
+@end
+
+/**
+ * An ORT session loads and runs a model.
+ */
+@interface ORTSession : NSObject
+
+/**
+ * Creates an ORT Session.
+ *
+ * @param env The ORT Environment instance.
+ * @param path The path to the ONNX model.
+ * @param error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (instancetype)initWithEnv:(ORTEnv*)env
+                  modelPath:(NSString*)path
```

Comment:
nit: can we just call this `init`? it seems that's possible based on [this](https://developer.apple.com/documentation/appkit/nsdocument) arbitrary document showing different ways to init an NSDocument

Reply:
I think we'll always need a model path. in Objective-C, a method's name is interspersed with its parameters (e.g. NSDocument's initWithContentsOfURL:ofType:error:).
in some other classes we could have an init that takes no parameters.

---
In reply to: [616199700](https://github.com/microsoft/onnxruntime/pull/7366#discussion_r616199700) [](ancestors = 616199700)

Reply:
will follow Apple naming conventions for methods with multiple parameters